test/common/testlib.py

```diff
@@ -122,14 +133,18 @@ def __init__(self, address, label, machine, pixels_label=None, port=None):
         self.default_user = "admin"
         self.label = label
         self.pixels_label = pixels_label
+        self.current_layout = "mobile" if bool(os.getenv("TEST_MOBILE", "")) else "desktop"
         self.machine = machine
         path = os.path.dirname(__file__)
-        self.cdp = cdp.CDP("C.utf8", verbose=opts.trace, trace=opts.trace, fixed_content_size=self.pixels_label,
+        self.cdp = cdp.CDP("C.utf8", verbose=opts.trace, trace=opts.trace,
+                           mobile=(self.current_layout == "mobile"),
```

Comment:
How about defining this as a boolean `is_mobile` property on the layout scenarios instead of hardcoding it here?  We might imagine several mobile scenarios with difference sizes, for example.

Reply:
Hmm, for this specific line, I think I would actually go back to the original code, which was setting `cdp.mobile` based on the TEST_MOBILE environment variable. `cdp.mobile` and TEST_MOBILE are now deprecated and I want to get rid of them eventually, once c-machines etc stop using them.

So keeping the old code would make most sense, I'd say.

But in general, I say it's a good idea to have more attributes in the layout definitions. If we move the definitions out of common/, then each project can put whatever stuff they want in there and check them in the tests.

But `is_mobile` will be a standard attribute, since the shell helpers in teslib.py itself need to know what layout the shell is in.

Reply:
> `cdp.mobile` and TEST_MOBILE are now deprecated and I want to get rid of them eventually, once c-machines etc stop using them.

Ah, let's just remove them altogether.  We don't need to keep testlib.py backwards compatible...